### ISSUE TITLE
Compile issues related to pkg/tsdb/mysql and pkg/api

### ISSUE THREAD

**Issue by dewi-ny-je** [grafana/grafana#8263]
The compilation stops with an error related to pkg/tsdb/mysql/mysql.go:

```
~/go/src/github.com/grafana/grafana olaf$ go run build.go setup
Version: 4.3.0-pre1, Linux Version: 4.3.0, Package Iteration: 1493673134pre1
go get -v github.com/kardianos/govendor
go install -v ./pkg/cmd/grafana-server
github.com/grafana/grafana/pkg/api
github.com/grafana/grafana/pkg/tsdb/mysql
# github.com/grafana/grafana/pkg/tsdb/mysql
pkg/tsdb/mysql/mysql.go:156: rows.ColumnTypes undefined (type *core.Rows has no field or method ColumnTypes)
pkg/tsdb/mysql/mysql.go:182: undefined: sql.ColumnType
# github.com/grafana/grafana/pkg/api
pkg/api/http_server.go:59: undefined: http.ErrServerClosed
pkg/api/http_server.go:65: undefined: http.ErrServerClosed
pkg/api/http_server.go:90: hs.httpSrv.Shutdown undefined (type *http.Server has no field or method Shutdown)
exit status 2
exit status 1
```

**Comment by daniellee** [comment]
Any clues on how to reproduce this. (It works on my machine :smiley: )

- What OS are you running on?
- Are you running the latest master?
- Did it work before?

**Comment by dewi-ny-je** [comment]
Sorry, I thought I attached the uname command.

I'm using OmniOS/illumos r151020 (current stable).
Go 1.7.1.

Go worked with influxdb, the compile environment seems good.
I was never able to get to this point before because first Grafana depended on govendor that was unable to compile. Solved that one, I had to manually update log15 (shipped with Grafana but very old).

I'm going step by step with my limited knowledge in my quest of a Grafana/illumos compilation.
